inst/shiny/modules/non_nca.R

```diff
@@ -0,0 +1,254 @@
+                  choices = plasma_options)
+    })
+
+
+    # Filter & prepare data for BPP calculation
+    filtered_samples_bpp <- reactive({
+      req(input$selected_blood, input$selected_plasmaforbpp)
+
+      data()$conc$data %>%
+        filter(PCSPEC %in% c(input$selected_blood, input$selected_plasmaforbpp))
+
+    })
+
+    # Perform BPP Calculation on Submit
+    bpp_results <- eventReactive(input$submit_bpp, {
+      req(filtered_samples_bpp())
+
+      blood <- input$selected_blood
+      plasma <- input$selected_plasmaforbpp
+
+      id_groups <- data()$conc$columns$groups %>%
+        purrr::list_c() %>%
+        append("DOSNO") %>%
+        purrr::keep(\(col) {
+          !is.null(col) && col != "PCSPEC" && length(unique(data()$conc$data[[col]])) > 1
+        })
+
+      bpp_groups <- c(grouping_vars(), id_groups,
+                      data()$dose$columns$dose, data()$dose$columns$time,
+                      data()$dose$columns$route)
+
+      # Separate Blood and Plasma Samples
+      df_blood <- filtered_samples_bpp() %>%
+        filter(PCSPEC == blood) %>%
+        rename(BLOOD_CONC = data()$conc$columns$concentration) %>%
+        select(bpp_groups, BLOOD_CONC)
+
+      df_plasma <- filtered_samples_bpp() %>%
+        filter(PCSPEC == plasma) %>%
+        rename(PLASMA_CONC = data()$conc$columns$concentration) %>%
+        select(bpp_groups, PLASMA_CONC)
+
+      # Merge Blood and Plasma Data
+      df_bpp <- left_join(df_blood, df_plasma, by = bpp_groups) %>%
+        filter(!is.na(BLOOD_CONC) & !is.na(PLASMA_CONC)) %>%
+        mutate(
+          BPP_RATIO = signif(BLOOD_CONC / PLASMA_CONC, 3)
+        ) %>%
+        select(bpp_groups, PLASMA_CONC, BLOOD_CONC, BPP_RATIO) %>%
+        arrange(USUBJID, TIME)
+
+      df_bpp
+    })
+
+    # Display results
+    output$bpp_results <- renderDT({
+      req(bpp_results())
+      datatable(bpp_results(), options = list(pageLength = 5))
+    })
+
+    #Tissue-Plasma Analysis ###############################
+    # Dynamically generate the tissue selection input
+    output$tissue_selector <- renderUI({
+      req(data())
+
+      tissue_options <- unique(data()$conc$data$PCSPEC)
+      selectInput(ns("selected_tissues"), "Choose Tissues",
+                  choices = tissue_options, multiple = TRUE,
+                  selected = tissue_options)
+    })
+
+    output$plasma_selector <- renderUI({
+      req(data())
+
+      tissue_options <- unique(data()$conc$data$PCSPEC)
+      selectInput(ns("selected_plasma"), "Choose Plasma",
+                  choices = tissue_options)
+    })
+
+    # Filter & prepare data for tissue-plasma ratio calculation
+    filtered_samples <- reactive({
+      req(data(), input$selected_tissues)
+
+      plasma <- input$selected_plasma
+      tissue <- input$selected_tissues
+
+      df_filtered <- data()$conc$data %>%
+        filter(PCSPEC %in% c(input$selected_plasma, input$selected_tissues))
+
+      df_filtered
```

Comment:
nitpick: No need to return the variable, if it is never assigned

```suggestion
      data()$conc$data %>%
        filter(PCSPEC %in% c(input$selected_plasma, input$selected_tissues))
```